### PR TITLE
rpc: glfs-operations: update api call for ftruncate

### PR DESCRIPTION
**Pull request by bhumikagoyal**
The updated gfapi for glusterfs fop ftruncate has pre/post attributes in the function signature. But this api call inside gluster-block is not updated, so update the api call for ftruncate to resolve build errors.

Signed-off-by: Bhumika Goyal <bgoyal@redhat.com>

### CODE REVIEW

File: rpc/glfs-operations.c [hunk 1]
```diff
@@ -174,7 +174,7 @@ glusterBlockCreateEntry(struct glfs *glfs, blockCreateCli *blk, char *gbid,
     ret = -1;
     goto out;
   } else {
-    ret = glfs_ftruncate(tgfd, blk->size);
+    ret = glfs_ftruncate(tgfd, blk->size, NULL, NULL);
```
Comment (pkalever) [inline rpc/glfs-operations.c]: Should we add version check for not breaking with older versions
#if CONFIG_GLUSTERFS_VERSION < X.Y.Z
 ret = glfs_ftruncate(tgfd, blk->size);
#else
ret = glfs_ftruncate(tgfd, blk->size, NULL, NULL);
#end

Comment (bhumikagoyal) [reply rpc/glfs-operations.c]: Yeah right. My change will break the build with older versions. Will put a check and update the PR.

File: rpc/glfs-operations.c [hunk 2]
```diff
@@ -8,7 +8,7 @@
   cases as published by the Free Software Foundation.
 */
 
-
+# include <../config.h>
```
Comment (amarts) [inline rpc/glfs-operations.c]: this doesn't look good! should be #include "config.h" or rather we should add '-include config.h' in CFLAGS in configure itself!

File: configure.ac [hunk 3]
```diff
@@ -80,8 +80,10 @@ AC_CHECK_HEADERS([stdio.h stdlib.h string.h stdbool.h stddef.h \
 
 # Checks for libraries.
 # glusterfs-api versions have a prefix of "7."
-PKG_CHECK_MODULES([GFAPI], [glusterfs-api >= 7.3.6],,
-                  [AC_MSG_ERROR([gfapi library >= 3.6 is required to build gluster-block])])
+
+PKG_CHECK_MODULES([GFAPI], [glusterfs-api >= 7.3.6],
+                  [PKG_CHECK_MODULES([gfapi], [glusterfs-api < 7.4.1],[AC_DEFINE([USE_LEGACY_FTRUNCATE],1,[Define 1 if glusterfs-api version < 7.4.1])],true)],
```
Comment (amarts) [inline configure.ac]: with release of glusterfs-5.0 these numbers may change... please have a look!

**Comment by pkalever** [comment]
As discussed at https://github.com/gluster/gluster-block/pull/150 we should close it.

**Comment by pkalever** [comment]
Fixed by: https://github.com/gluster/gluster-block/pull/178

**Pull request closed without merging**